Python/Product/Debugger/DebugAdapterProcess.cs

```diff
@@ -38,21 +38,31 @@ public sealed class DebugAdapterProcess : ITargetHostProcess {
         private string _interpreterOptions;
         private int _listenerPort = -1;
         private Stream _stream;
-        private Socket _socket;
-        private ManualResetEvent _connectedEvent = new ManualResetEvent(false);
 
         public DebugAdapterProcess() {
             _processGuid = Guid.NewGuid();
         }
 
+        public void Dispose() {
+            if (_stream != null) {
+                _stream.Dispose();
+            }
+            if (_process != null) {
+                if (!_process.HasExited) {
+                    _process.Kill();
```

Comment:
This can throw (e.g. if there's a race between `HasExited` and `Kill`). Since we're in `Dispose`, probably a good idea to catch/discard.